OFEP-ofo-flagd-client-support.md

```diff
@@ -0,0 +1,36 @@
+## flagd client support driven by OpenFeature Operator
+
+## State: DRAFTING
+
+Client side applications (e.g. web/mobile) could benefit from access to feature flags. This OFEP outlines the feasibility and benefits of extending [OpenFeature Operator (OFO)](https://github.com/open-feature/open-feature-operator) to manage the deployment of flagd with access from external traffic (e.g. client side applications).
+
+## Background
+
+OFO already manages the deployment of flagd in server side contexts by injecting it as a sidecar container to an existing workload. This permits the workload to communicate with flagd due to the inherent nature of networking between containers within a pod. Conversely, the routing of client side applications to flagd is not trivial. OFO could bear this burden by configuring the cluster as necessary (see the proposal below) to facilitate the deployment of externally accessible (by client side applications) flagd.
+
+## Assumptions
+The (simplified) deployment pattern is as follows
+
+- Deployment of a client side application exposed by a Service
+- [Gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/) exposing the app's Service via a [HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/) and listener
+
+The networking infrastructure for ingesting external traffic is already in place for the existing client side application.
+
+## Proposal
+
+Introduce a custom resource definition (CRD) with configuration permitting OFO to create a deployment of flagd and inject the necessary networking configuration to route incoming traffic from the existing Gateway to flagd (via a new HTTPRoute/GRPCRoute).
+This results in flagd being externally accessible.
+
+The following diagram depicts the architecture. Resources inside the green box are created via the proposed CRD.
+
+![OFO client architecture](images/ofo-flagd-client-support-architecture.png "OFO Client Architecture")
+
+Note: The example depicted by the diagram uses a subdomain to route the request to flagd but [path matching is also possible](https://gateway-api.sigs.k8s.io/api-types/httproute/#matches).
+
+## Limitations
+[Kubernetes Gateway API (KGA)](https://gateway-api.sigs.k8s.io/) is in beta (as are its implementors). This OFEP could be extended (or a new one created) to also support [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/).
+The [GRPCRoute](https://gateway-api.sigs.k8s.io/api-types/grpcroute/) is experimental.
```

Comment:
It seems like the `GRPCRoute` and maybe even the `HTTPRoute` (if we are using connect-web/connect-es) might allow us to map requests?

As in, if a use has a web-app hosted at origin `app.com`, it might enable us to map only requests to `app.com/flags` to flagd. It would be great if we could get this to work so that users wouldn't be forced to use a separate subdomain for flagd requests (flags.app.com).